applications/RomApplication/python_scripts/rom_manager.py

```diff
@@ -256,280 +228,352 @@ def RunHROM(self, mu_run=[None], use_full_model_part = False):
-            parameters_copy = self._AddBasisCreationToProjectParameters(parameters_copy) #TODO stop using the RomBasisOutputProcess to store the snapshots. Use instead the upcoming build-in function
-            parameters_copy = self._StoreResultsByName(parameters_copy,'FOM_Fit',mu,Id)
-            materials_file_name = parameters_copy["solver_settings"]["material_import_settings"]["materials_filename"].GetString()
-            self.UpdateMaterialParametersFile(materials_file_name, mu)
-            model = KratosMultiphysics.Model()
-            analysis_stage_class = self._GetAnalysisStageClass(parameters_copy)
-            simulation = self.CustomizeSimulation(analysis_stage_class,model,parameters_copy)
-            simulation.Run()
-            self.QoI_Fit_FOM.append(simulation.GetFinalData())
-            for process in simulation._GetListOfOutputProcesses():
-                if isinstance(process, CalculateRomBasisOutputProcess):
-                    BasisOutputProcess = process
-            SnapshotsMatrix.append(BasisOutputProcess._GetSnapshotsMatrix()) #TODO add a CustomMethod() as a standard method in the Analysis Stage to retrive some solution
-        SnapshotsMatrix = np.block(SnapshotsMatrix)
-        u, sigma = BasisOutputProcess._ComputeSVD(SnapshotsMatrix)
-        BasisOutputProcess._PrintRomBasis(u, sigma) #Calling the RomOutput Process for creating the RomParameter.json
+        self._LaunchTrainFOM(mu_train)
+        self._LauchComputeSolutionBasis(mu_train)
+
+
+
+    def _LaunchTrainFOM(self, mu_train):
+        in_database, hash_basis = self.data_base.check_if_in_database("RightBasis", mu_train)
+        if not in_database:
+            with open(self.project_parameters_name,'r') as parameter_file:
+                parameters = KratosMultiphysics.Parameters(parameter_file.read())
+            for Id, mu in enumerate(mu_train):
+                in_database, _ = self.data_base.check_if_in_database("FOM_Fit", mu)
+                if not in_database:
+                    parameters_copy = self.UpdateProjectParameters(parameters.Clone(), mu)
+                    parameters_copy = self._AddBasisCreationToProjectParameters(parameters_copy) #TODO stop using the RomBasisOutputProcess to store the snapshots. Use instead the upcoming build-in function
+                    parameters_copy = self._StoreResultsByName(parameters_copy,'FOM_Fit',mu,Id)
+                    materials_file_name = parameters_copy["solver_settings"]["material_import_settings"]["materials_filename"].GetString()
+                    self.UpdateMaterialParametersFile(materials_file_name, mu)
+                    model = KratosMultiphysics.Model()
+                    analysis_stage_class = self._GetAnalysisStageClass(parameters_copy)
+                    simulation = self.CustomizeSimulation(analysis_stage_class,model,parameters_copy)
+                    simulation.Run()
+                    self.QoI_Fit_FOM.append(simulation.GetFinalData())
```

Comment:
I'm not sure how this QoI_Fit_FOM is used, but in here it is only getting results for the snapshots that were not in the database already.